aws-push-notifications-pinpoint/src/main/java/com/amplifyframework/pushnotifications/pinpoint/FCMPushNotificationsService.kt

```diff
@@ -42,11 +43,16 @@ internal class FCMPushNotificationsService : FirebaseMessagingService() {
         // * The wakelock ID set by the WakefulBroadcastReceiver
         data.remove("androidx.content.wakelockid")
 
-        // get pinpoint notifications payload
-        val notificationPayload = PinpointNotificationPayload.createFromRemoteMessage(RemoteMessage(data))
+        // create notifications payload
+        val remoteMessage = RemoteMessage((data))
+        val notificationPayload = NotificationPayload {
+            notificationContentProvider(NotificationContentProvider.FCM(remoteMessage.from, remoteMessage.data))
+        }
+
+        val isAmplifyMessage = Amplify.Notifications.Push.shouldHandleNotification(notificationPayload)
```

Comment:
only 2 cases may be better to use if/else instead of when

Reply:
yes, forgot to change this. It was if in the original PR.